src/components/pagination/daikin-pagination.ts

```diff
@@ -0,0 +1,232 @@
+  "focus-visible:outline-[#0081C0]",
+]);
+
+/**
+ * The pagination component is used to navigate through a list of items that are divided into multiple pages.
+ *
+ * @fires change - Emitted when the pagination current is changed.
+ *
+ * @example
+ *
+ * ```html
+ * <!-- You can create a pagination just specify the max pages and how many pages should be show. -->
+ * <daikin-pagination total="5" window="5">
+ * </daikin-pagination>
+ * ```
+ */
+@customElement("daikin-pagination")
+export class DaikinPagination extends LitElement {
+  static override readonly styles = css`
+    ${unsafeCSS(tailwindStyles)}
+
+    :host {
+      display: inline-flex;
+    }
+  `;
+
+  /**
+   * The current page number, starting at 1.
+   * Must be greater than 0 and less than or equal to `this.total`.
+   */
+  @property({ type: Number, reflect: true })
+  current = 1;
+
+  /**
+   * The number of pages.
+   * Must be greater than 0.
+   */
+  @property({ type: Number, reflect: true })
+  total = 5;
+
+  /**
+   * Number of elements to display in pagination, including ellipses.
+   * Must be greater than or equal to 5.
+   * If a value less than 5 is specified, it will be treated as 5.
+   */
+  @property({ type: Number, reflect: true })
+  window = 5;
+
+  private _goto(page: number): void {
+    this.current = Math.max(page, 1);
+    this.dispatchEvent(
+      new Event("change", {
+        bubbles: true,
+        composed: true,
+        cancelable: true,
+      })
+    );
+  }
+
+  private _gotoOffset(offset: number): void {
+    this._goto(this.current + offset);
+  }
+
+  override render() {
+    const cvaChevron = cvaChevronButton();
+
+    const pageArray = calculatePagination(
+      this.total,
+      this.current,
+      this.window
+    );
+    return html`
+      <div aria-label="Pagination" class="inline-flex gap-1">
+        <button
+          class=${cvaChevron}
+          type="button"
+          aria-label="Go to the previous page."
+          ?disabled=${this.current === 1}
+          @click=${() => this._gotoOffset(-1)}
+        >
+          <span class="i-daikin-pagination-chevron-left w-4 h-4"></span>
+        </button>
+
+        ${repeat(pageArray, (pageItem) => {
+          const ellipsisClassName = cvaEllipsis();
+          if (pageItem.type === "page") {
+            const buttonClassName = cvaPageButton({
+              active: this.current === pageItem.page,
+            });
+            return html`
+              <button
+                type="button"
+                class=${buttonClassName}
+                @click=${() => this._goto(pageItem.page)}
+                aria-label="Go to page ${pageItem.page}"
+              >
+                ${pageItem.page}
+              </button>
+            `;
+          } else {
+            const omittedPages = pageItem.pages;
+            if (omittedPages.length > 0) {
+              return html`<div class="relative">
+                <span class=${ellipsisClassName}>
```

Comment:
```suggestion
                <span class=${cvaEllipsis()}>
```

Maybe we should also do `cvaEllipsis()` in outside the class.